__init__.py

```diff
@@ -15,95 +15,406 @@
-                assert self.repeat_data
-        except:
-            self.repeat_data = {}
-
-    def __handle_stop(self, message):
-        if self.alarm_on:
-            self.speak_dialog('alarm.off')
-        self.alarm_on = False
-
-    def notify(self, timestamp):
-        with self.LOCK:
-            if self.data.__contains__(timestamp):
-                volume = None
-                self.alarm_on = True
-                delay = self.__calculate_delay(self.max_delay)
-
-                while self.alarm_on and datetime.now() < delay:
-                    play_mp3(self.file_path).communicate()
-                    self.speak_dialog('alarm.stop')
-                    time.sleep(self.repeat_time + 2)
-                    if not volume and datetime.now() >= delay:
-                        mixer = Mixer()
-                        volume = mixer.getvolume()[0]
-                        mixer.setvolume(100)
-                        delay = self.__calculate_delay(self.extended_delay)
-                if volume:
-                    Mixer().setvolume(volume)
-                self.remove(timestamp)
-                self.alarm_on = False
-                self.save()
-
-    @staticmethod
-    def __calculate_delay(seconds):
-        return datetime.now() + timedelta(seconds=seconds)
-
-    def save(self):
-        with self.file_system.open(self.PENDING_TASK, 'w') as f:
-            yaml.safe_dump(self.data, f)
-        with self.file_system.open(self.REPEAT_TASK, 'w') as f:
-            yaml.safe_dump(self.repeat_data, f)
-        if not self.alarm_on:
-            self.schedule()
+        # self.register_intent_file('set.morning.intent', self.set_morning)
+        # self.register_intent_file('set.sunrise.intent', self.set_sunrise)
+        # self.register_intent_file('delete.all.intent', self.delete_all)
+        # self.register_entity_file('exceptdaytype.entity')
+
+        # using adapt because padatious seems to
+        # not pick this up for some reason
+        status_intent = IntentBuilder("status.alarm.intent"). \
+            require("status").require("alarm").build()
+        self.register_intent(status_intent, self.handle_status)
+
+        # using padatious
+        self.register_intent_file('delete.intent', self.handle_delete)
+        self.register_intent_file('set.recurring.intent',
+                                  self.handle_set_recurring)
+        self.register_intent_file('stop.intent', self.stop)
+        if self.time_format == 'half':
+            self.register_intent_file('set.time.intent', self.handle_set_time)
+            self.register_entity_file('ampm.entity')
+        elif self.time_format == 'full':
+            self.register_intent_file('set.time.24hr.intent',
+                                      self.handle_set_time)
+        self.register_entity_file('time.entity')
+        self.register_entity_file('length.entity')
+        self.register_entity_file('daytype.entity')
```

Comment:
We should change all these to the decorator form.

Reply:
@MatthewScholefield what does the decorator form look like?

Reply:
[Here's the documentation](https://github.com/MycroftAI/mycroft-core/wiki/Intent-handler-decorators#using-intent-handler-decorators). For Padatious it's `@intent_file_handler(...`.